codeboarding_web/app.py

```
@@ -0,0 +1,222 @@
+        return JSONResponse(elements)
+
+    @app.get("/api/component/{component_id}/diff")
+    def component_diff_route(component_id: str) -> JSONResponse:
+        """Return git diff patch for the files belonging to a component, or 404."""
+        path = output_dir / "analysis.json"
+        if not path.exists():
+            raise HTTPException(status_code=404, detail="no analysis yet")
+        try:
+            data = json.loads(path.read_text(encoding="utf-8"))
+            root_analysis, sub_analyses = parse_unified_analysis(data)
+        except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)